src/main/java/com/shaft/tools/io/YAMLFileManager.java

```diff
@@ -0,0 +1,404 @@
+     * @param clazz the class that data wanted to be parsed for
+     * @return the wanted value as {@link Map} of {@link String} and {@link T}
+     */
+    public <T> Map<String, T> getMapAs(String key, Class<T> clazz) {
+        return parseObjectToMap(get(key), clazz);
+    }
+
+    /**
+     * Used internally to fetch a value form a map using a list of keys
+     *
+     * @param map the map wanted to fetch data form
+     * @param keys the list of keys that point on the value
+     * @return the wanted value as {@link Object}
+     */
+    private Object getValueFromKeys(Map<String, Object> map, List<String> keys) {
+        Object value = map;
+        for (int i = 0; i < keys.size(); i++) {
+            var isLastItem = i == keys.size() - 1;
+            if (keyHasList.test(keys.get(i))) {
+                value = getValueFromKeyList(
+                        parseObjectToMap(value, Object.class),
+                        keys.get(i));
+
+                if (isLastItem) break;
+
+            } else
+                value = parseObjectToMap(value, Object.class)
+                        .get(keys.get(i));
+
+            if (!isLastItem)
+                value = parseObjectToMap(value, Object.class);
+        }
+
+        return value;
+    }
+
+    /**
+     * Used internally to fetch data from keys that have an index for a list in it
+     * @param map the map wanted to fetch data form
+     * @param key the key that has the index on it
+     * @return the wanted value as {@link Object}
+     */
+    private Object getValueFromKeyList(Map<String, Object> map, String key) {
+        var indexes = parseKeyList(key);
+        key = key.replaceAll(NUMBER_IN_SQUARE_BRACKETS_REGEX, "");
+
+        var value = map.get(key);
+
+        for (int i = 0; i < indexes.size(); i++) {
+            value = parseObjectToList(value, Object.class)
+                    .get(indexes.stream().findFirst().orElseThrow());
+        }
+
+        return value;
+    }
+
+    /**
+     * Used internally to fetch all data existed in the YAML file
+     * @return all data in the YAML file as {@link Map} of {@link String} and {@link Object}
+     */
+    private Map<String, Object> getData() {
+        var file = this.getFile();
+        Map<String, Object> loadedData = new Yaml().load(file);
+
+        this.closeFile(file);
+
+        return loadedData;
+    }
+
+    /**
+     * Used internally to load the wanted YAML file
+     * @return {@link FileInputStream} instance of the desired file
+     */
+    private FileInputStream getFile() {
+        FileInputStream in;
+
+        try {
+            in = new FileInputStream(filePath);
+        } catch (FileNotFoundException rootCauseException) {
+            var errorMsg = "Couldn't find the desired file. [" + filePath + "].";
+            sendFailureCompo(errorMsg, rootCauseException);
+            throw new RuntimeException(errorMsg);
```

Comment:
Note that in case of exceptions that should fail the execution we should follow this pattern for consistent reporting:
https://github.com/ShaftHQ/SHAFT_ENGINE/blob/b90e8ff225776bc014f6f57680fc3bf67f646e24/src/main/java/com/shaft/tools/io/JSONFileManager.java#L119-L122